salt/state.py

```diff
@@ -757,7 +758,21 @@ def __init__(
         loader="states",
         initial_pillar=None,
         file_client=None,
+        __invocation_id=None,
     ):
+        """
+        When instantiating an object of this class, do not pass
+        ``__invocation_id``. It is an internal field for tracking
+        parallel executions where no jid is available (Salt-SSH) and
+        only exposed as an init argument to work on spawning platforms.
+        """
+        if jid is not None:
+            __invocation_id = jid
```

Comment:
I quickly applied this, but then noticed this will likely break parallel execution on spawning platforms: Each subprocess then generates its own `__invocation_jid` since `jid` is always `None`, but we need it to be the same to find the cached return.

The tests will not catch this though since SSH tests only run against forking platforms.

Reply:
Okay, makes sense.